### PR TITLE
Use logger conditionally wherever possible

### LINKED ISSUE #589: Chewy crashes when `config.active_record.logger = nil`
As explains in [rails guide](http://guides.rubyonrails.org/configuring.html#configuring-active-record) I set `config.active_record.logger` to `nil` to disable active record log.

But that crashes the app during [Chewy logger call](https://github.com/toptal/chewy/blob/master/lib/chewy/railtie.rb#L17)

```shell
Puma caught this error: undefined method `info' for nil:NilClass (NoMethodError)
/home/mickro/.rvm/gems/ruby-2.4.2/gems/chewy-0.10.1/lib/chewy/railtie.rb:17:in `call'
/home/mickro/.rvm/gems/ruby-2.4.2/gems/railties-5.0.5/lib/rails/rack/logger.rb:36:in `call_app'
/home/mickro/.rvm/gems/ruby-2.4.2/gems/railties-5.0.5/lib/rails/rack/logger.rb:24:in `block in call'
/home/mickro/.rvm/gems/ruby-2.4.2/gems/activesupport-5.0.5/lib/active_support/tagged_logging.rb:69:in `block in tagged'
/home/mickro/.rvm/gems/ruby-2.4.2/gems/activesupport-5.0.5/lib/active_support/tagged_logging.rb:26:in `tagged'
/home/mickro/.rvm/gems/ruby-2.4.2/gems/activesupport-5.0.5/lib/active_support/tagged_logging.rb:69:in `tagged'
/home/mickro/.rvm/gems/ruby-2.4.2/gems/railties-5.0.5/lib/rails/rack/logger.rb:24:in `call'
/home/mickro/.rvm/gems/ruby-2.4.2/gems/sprockets-rails-3.2.0/lib/sprockets/rails/quiet_assets.rb:13:in `call'
/home/mickro/.rvm/gems/ruby-2.4.2/gems/request_store-1.3.2/lib/request_store/middleware.rb:9:in `call'
/home/mickro/.rvm/gems/ruby-2.4.2/gems/actionpack-5.0.5/lib/action_dispatch/middleware/request_id.rb:24:in `call'
/home/mickro/.rvm/gems/ruby-2.4.2/gems/rack-2.0.3/lib/rack/method_override.rb:22:in `call'
/home/mickro/.rvm/gems/ruby-2.4.2/gems/rack-2.0.3/lib/rack/runtime.rb:22:in `call'
/home/mickro/.rvm/gems/ruby-2.4.2/gems/activesupport-5.0.5/lib/active_support/cache/strategy/local_cache_middleware.rb:28:in `call'
/home/mickro/.rvm/gems/ruby-2.4.2/gems/actionpack-5.0.5/lib/action_dispatch/middleware/executor.rb:12:in `call'
/home/mickro/.rvm/gems/ruby-2.4.2/gems/actionpack-5.0.5/lib/action_dispatch/middleware/static.rb:136:in `call'
/home/mickro/.rvm/gems/ruby-2.4.2/gems/rack-2.0.3/lib/rack/sendfile.rb:111:in `call'
/home/mickro/.rvm/gems/ruby-2.4.2/gems/railties-5.0.5/lib/rails/engine.rb:522:in `call'
/home/mickro/.rvm/gems/ruby-2.4.2/gems/puma-3.10.0/lib/puma/configuration.rb:225:in `call'
/home/mickro/.rvm/gems/ruby-2.4.2/gems/puma-3.10.0/lib/puma/server.rb:605:in `handle_request'
/home/mickro/.rvm/gems/ruby-2.4.2/gems/puma-3.10.0/lib/puma/server.rb:437:in `process_client'
/home/mickro/.rvm/gems/ruby-2.4.2/gems/puma-3.10.0/lib/puma/server.rb:301:in `block in run'
/home/mickro/.rvm/gems/ruby-2.4.2/gems/puma-3.10.0/lib/puma/thread_pool.rb:120:in `block in spawn_thread'
` ``

### PR DESCRIPTION
**Pull request by pyromaniac**
Fixes #589 #581

### CODE REVIEW

File: lib/chewy/railtie.rb [hunk 1]
```diff
@@ -43,14 +43,12 @@ def migrate(*args)
     end
 
     console do |app|
-      Chewy.logger = ActiveRecord::Base.logger if defined?(ActiveRecord)
```
Comment (mlen) [inline lib/chewy/railtie.rb]: I think it'd be cleaner this way:
```
Chewy.logger = defined?(ActiveRecord) ? ActiveRecord::Base.logger : Logger.new(STDERR)
```

Comment (pyromaniac) [reply lib/chewy/railtie.rb]: No need to do it, rails does the same, https://github.com/rails/rails/blob/master/railties/lib/rails/commands/console/console_command.rb#L58, just puts

Comment (mlen) [reply lib/chewy/railtie.rb]: Well, the fact that they do the same doesn't change anything. If you assign a your own logger instance when AR is not available, you'll at least be able to get rid of that ugly conditional above

Comment (pyromaniac) [reply lib/chewy/railtie.rb]: I understand your point, I was thinking about this but decided to keep it simple now.

**Comment by maxpollack** [comment]
Is this fix in the mainline currently? I'm experiencing this same issue currently with a fresh install of the `chewy` gem.

**Pull request merged**